### PR TITLE
fix: add missing Properties

### PR DESCRIPTION
**Pull request by desenna**
Closes https://github.com/CycloneDX/cyclonedx-go/issues/38

### CODE REVIEW

File: cyclonedx_test.go [hunk 1]
```diff
@@ -217,3 +217,11 @@ func TestLicenses_UnmarshalXML(t *testing.T) {
 	err = xml.Unmarshal([]byte("<Licenses><somethingElse>expressionValue</somethingElse></Licenses>"), licenses)
 	assert.Error(t, err)
 }
+
+func TestVulnerability_Properties(t *testing.T) {
+	// GIVEN
+	var vuln Vulnerability
+
+	// EXPECT
+	assert.Equal(t, 0, len(*vuln.Properties))
```
Comment (nscuro) [inline cyclonedx_test.go:226]: This will always panic, because `vuln.Properties` is `nil`.

**Review by nscuro — CHANGES_REQUESTED** [review]
Thanks for the PR @desenna! 

Besides `TestVulnerability_Properties` currently panicking, the snapshot files for `valid-vulnerability.json` and `valid-vulnerability.xml` have to be updated.

You can do that by running

```shell
UPDATE_SNAPSHOTS=true make test
```

**Comment by nscuro** [comment]
Welp, looks like you discovered a bug in the XML schema! 🚀

I filed a bug for this in the specification repo: https://github.com/CycloneDX/specification/issues/147

Unfortunate, but it may take some time for the schema to be updated. For now you could comment out the properties in `valid-vulnerability.xml` (but not the JSON file) and regenerate the snapshots. That should make the test pass.

**Comment by desenna** [comment]
Yep, discovered a bug. Just wanted to check with my teammate that I wasn't crazy. Ok, going to add new commit now.

**Comment by desenna** [comment]
It points to lack of tests.

File: testdata/valid-vulnerability.xml [hunk 2]
```diff
@@ -116,6 +116,12 @@
                     </versions>
                 </target>
             </affects>
+            <!-- <properties>
```
Comment (desenna) [inline testdata/valid-vulnerability.xml:119]: TODO: https://github.com/CycloneDX/cyclonedx-go/issues/40

**Review by nscuro — APPROVED** [review]

**Comment by desenna** [comment]
Thanks for merging @nscuro . Any idea when this can be released as a v0.5.3 so we can use it? If easy to do, would it be possible to release today so I don't have to code a workaround on my side and then undo it? 😉

**Comment by nscuro** [comment]
I'm on the go currently but I'll most likely release v0.6.0 later today. Give me 1-2 hours.

**Comment by nscuro** [comment]
@desenna Just pushed v0.6.0. Thanks again for your PR!

**Comment by desenna** [comment]
> @desenna Just pushed v0.6.0. Thanks again for your PR!

Thanks @nscuro !!

**Pull request merged**